hedyorg/hedy
# [UI] add teacher preview mode

**Description**

Adds a preview mode for teachers, so when they create a class they can view Hedy as if they were a student in this class.

Fixes #3374

This is a draft PR, please let me know if this is roughly the right direction. Once we confirm that I'll add:
- some tests
- make the banner "Preview Mode" look nicer
- add translation keys

**How to test**

Step 1 - Go to For Teacher page. Make sure you have a class with some customizations, like less programs for level 1. Press the "Preview" button.

![Screenshot from 2023-11-29 12-09-57](https://github.com/hedyorg/hedy/assets/1298565/24a1056a-f4e7-4910-a2fa-a37a609a9d39)

Step 2 - You should go to /hedy and see the loaded class including a banner "Preview mode".

![Screenshot from 2023-12-18 13-45-47](https://github.com/hedyorg/hedy/assets/1298565/c7ac4b5e-314b-487a-9023-866e63190ba1)


The previous version of this design is [here](https://github.com/hedyorg/hedy/assets/1298565/a13ac70d-5524-435a-884a-061164e3160d).

Step 3 - You can complete the programs and puzzles as if you were a student

Step 4 - When you exit preview mode the banner should disappear and you should go back to the /for-teachers page.


**Checklist**
Done? Check if you have it all in place using this list: (mark with x if done)
